### PR TITLE
feat(nextcloud): Implement WebDAV extended MKCOL

### PR DESCRIPTION
**Pull request by provokateurin**
Closes https://github.com/nextcloud/neon/issues/512

### CODE REVIEW

File: packages/nextcloud/lib/src/api/webdav/webdav_client.dart [hunk 1]
```diff
@@ -62,23 +62,43 @@ class WebDavClient {
...
   ///   * [mkcol] for a complete operation executing this request.
-  http.Request mkcol_Request(PathUri path) {
+  http.Request mkcol_Request(
+    PathUri path, {
+    WebDavProp? set,
+  }) {
     final request = http.Request('MKCOL', _constructUri(path));
+    if (set != null) {
+      request
+        ..encoding = utf8
```
Comment (Leptopoda) [inline packages/nextcloud/lib/src/api/webdav/webdav_client.dart:77]: utf8 is already set by default

**Review by Leptopoda — APPROVED** [review]
I have no Idea what this is used for but the changes look sane

File: packages/nextcloud/lib/src/api/webdav/webdav_client.dart (hunk 1 above)
Comment (provokateurin) [reply packages/nextcloud/lib/src/api/webdav/webdav_client.dart:77]: Copy-pasta

**Comment by provokateurin** [comment]
> I have no Idea what this is used for but the changes look sane

You can simply set props while creating resources (mostly directories). It saves you another request.

**Pull request merged**